Modules/posixmodule.c

```diff
@@ -1734,6 +1734,39 @@ posix_fildes_fd(int fd, int (*func)(int))
 
 
 #ifdef MS_WINDOWS
+static wchar_t *
+win32_wgetcwd(wchar_t *buf, DWORD buf_size)
+{
+    wchar_t *local_buf = buf;
+    while (1) {
+        wchar_t *temp;
+        DWORD result = GetCurrentDirectoryW(buf_size, local_buf);
+        if (!result)
+            goto fail;
+
+        /* L'\0' is not counted in result on success. */
+        if (result < buf_size)
+            break;
+
+        buf_size = result;
+        temp = PyMem_RawRealloc(local_buf != buf ? local_buf : NULL,
+                                buf_size * sizeof(wchar_t));
+        if (!temp) {
+            SetLastError(ERROR_OUTOFMEMORY);
```

Comment:
```suggestion
            SetLastError(ERROR_NOT_ENOUGH_MEMORY);
```

The error code `ERROR_OUTOFMEMORY` is rarely seen in the Windows API, and only occurs when hard coded, likely by a Microsoft developer who picked the first error code that looked right instead of researching the error that was in common use. The automatic mapping from NT is `RtlNtStatusToDosError(STATUS_NO_MEMORY)` -> `ERROR_NOT_ENOUGH_MEMORY`. That's also the error code that the C runtime maps to the errno value `ENOMEM`, which Python defines as well in [`winerror_to_errno()`](https://github.com/python/cpython/blob/d8c7a1174cc182668085b10aab4049f6a2794c2f/PC/errmap.h#L71-L75) in "PC/errmap.h".

Reply:
Thanks for catching that. I thought I'd researched it when doing the original PR.